File: watcher/ingest/raw-record.ts
export interface RawStationRecord {
  readonly id: number;
  readonly geom?: { readonly lon: number; readonly lat: number } | null;
  readonly latitude?: string | number | null;
  readonly longitude?: string | number | null;
  readonly adresse?: string | null;
  readonly ville?: string | null;
  readonly cp?: string | null;
  readonly [field: string]: unknown;
}

File: src-tauri/src/ingest/raw_record.rs
use serde::Deserialize;
use serde_json::Value;
use std::collections::HashMap;

#[derive(Debug, Clone, Default, Deserialize)]
pub struct Geom {
    #[serde(default)]
    pub lat: Option<f64>,
    #[serde(default)]
    pub lon: Option<f64>,
}

#[derive(Debug, Clone, Default, Deserialize)]
pub struct RawStationRecord {
    #[serde(default)]
    pub id: i64,
    #[serde(default)]
    pub geom: Option<Geom>,
    #[serde(default)]
    pub latitude: Option<Value>,
    #[serde(default)]
    pub longitude: Option<Value>,
    #[serde(default)]
    pub adresse: Option<Value>,
    #[serde(default)]
    pub ville: Option<Value>,
    #[serde(flatten)]
    pub extra: HashMap<String, Value>,
}
